server/test/unit/server/webservices/views/test_task_groups.py

```diff
@@ -27,6 +28,55 @@ def filter(self, state=None):
         return MockQuerySet(filtered_list)
 
 
+class TestTaskGroupView(unittest.TestCase):
+    """
+    Tests for TaskGroupView
+    """
+    @mock.patch('pulp.server.webservices.views.decorators._verify_auth',
+                new=assert_auth_DELETE())
+    @mock.patch('pulp.server.webservices.views.task_groups.TaskStatus.objects')
+    @mock.patch('pulp.server.webservices.views.task_groups.tasks')
+    def test_delete_task_resource_nonexistant(self, mock_task, mock_objects):
+        """
+        Test delete task_group with no tasks
+        """
+        mock_request = mock.MagicMock()
+        mock_objects.only.return_value.get_or_404.side_effect = MissingResource
+        task_resource = TaskGroupView()
+        try:
+            response = task_resource.delete(mock_request, 'mock_task')
+        except MissingResource, response:
```

Comment:
You re-used the variable ``response`` to hold an exception instead of a response object. It's better to use a different variable name.